lms/djangoapps/courseware/module_render.py

```diff
@@ -165,6 +171,10 @@ def toc_for_course(request, course, active_chapter, active_section, field_data_c
                 if unicode(chapter.location) not in required_content:
                     local_hide_from_toc = True
 
+            # check if user is allowed to skip entrance exam then don't hide content
```

Comment:
If we move the check above into _get_required_content() does this check/override become unnecessary?